benchmark/benchmark.hpp

```diff
@@ -83,6 +85,82 @@ struct alloc_data {
     size_t size;
 };
 
+struct next_alloc_data {
+    size_t offset;
+    size_t size;
+};
+
+#ifndef WIN32
+std::vector<cpu_set_t> affinityMask;
+
+int initAffinityMask() {
+    cpu_set_t mask;
+    CPU_ZERO(&mask);
+
+    if (sched_getaffinity(0, sizeof(mask), &mask) == -1) {
+        perror("sched_getaffinity");
+        return 1;
+    }
+
+    for (int cpu = 0; cpu < CPU_SETSIZE; cpu++) {
```

Comment:
Avoid using cpu 0, it's likely to be the most occupied one of all.
This should be a configurable option.

Reply:
It is somehow configurable. I use sched_getaffinity to get list of available CPUs. If you limit benchmark to single node, or specific CPUs it will handle it. 

But you are right. I will reverse this list so we go from highest cpu